src/streaming-client.ts

```diff
@@ -0,0 +1,97 @@
+import * as fs from 'fs';
+import * as sockets from 'websocket';
+import * as events from 'events';
+import { Duplex, PassThrough } from 'stream';
+
+import AudioConfig from './models/streaming/AudioConfig';
+import BufferedDuplex from './models/streaming/BufferedDuplex';
+import StreamingResponse from './models/streaming/StreamingResponse';
+
+export class RevAiStreamingClient extends events.EventEmitter {
+    accessToken: string;
+    config: AudioConfig;
+    baseUrl: string;
+    requests: PassThrough;
+    responses: PassThrough;
+    client: sockets.WebSocketClient;
+    constructor(accessToken: string, config: AudioConfig, version = 'v1alpha'){
+        super();
+        this.accessToken = accessToken;
+        this.config = config;
+        this.baseUrl = `wss://api.rev.ai/speechtotext/${version}/stream?` +
+            `access_token=${accessToken}` +
+            `&content_type=${config.getContentTypeString()}`;
```

Comment:
I think we should either do some validation `AudioConfig` client side, or at least provide defaults. What do you think?

Reply:
also `access_token`, right now the backend returns a 401 (or 403?) http status code on unauthorized. Can we support returning the proper error in this client?

Reply:
I think defaulting to `audio/*` is reasonable. I don't think doing validation is reasonable for the same reason we don't validate on the midtier. It would be nice in order to fail fast but it isn't the responsibility of this sdk imo

Reply:
👍 Don't do validation on the client side as this duplicates validation logic and slows down our development.